src/main/scala/is/hail/annotations/ReadRowsRDD.scala

```diff
@@ -312,54 +392,81 @@ class RichRDDUnsafeRow(val rdd: RDD[UnsafeRow]) extends AnyVal {
 
 case class ReadRowsRDDPartition(index: Int) extends Partition
 
+object ReadRowsRDD {
+  def readInt(in: InputStream): Int = {
+    val b0 = in.read()
+    val b1 = in.read()
+    val b2 = in.read()
+    val b3 = in.read()
+
+    (b0 & 0xFF) |
```

Comment:
[`InputStream.read`](https://docs.oracle.com/javase/8/docs/api/java/io/InputStream.html#read--) ensures the high  bits are zero unless the end of stream is reached (well, really, it says the value will be between 0 and 255, and AFAIK, java guarantees two's complement representation). Is the `& 0xff` necessary? In the case that the stream terminates half-way through an int, no sensible value could be returned anyway, so it doesn't seem bad for the returned Int to also be `-1`.